tests/trace/type_handlers/Video/video_edge_cases_test.py

```diff
@@ -22,26 +23,26 @@ def sample_mp4_path():
         tmp_path, codec="libx264", audio=False, verbose=False, logger=None
     )
 
-    return tmp_path
+    yield tmp_path
```

Comment:
you have to yield inside the context manager, otherwise the file will get cleaned up before it gets to the test

Reply:
I wonder why the tests are still passing with it like this. That might indicate it's not working how we think it is

Reply:
Fixed, the file was being written after the tempfile was closed. I put both the writing and the yield inside and now it works properly. Nice catch.